metaseq/data/streaming_src_tgt_dataset.py

```diff
@@ -176,34 +181,50 @@ def yield_src_tgt_single_sentences_pad_8(iterable, block_size, drop_last, paddin
     return the example as is, without packing, truncating to block_size in cases of
     very long examples.
     """
+    counter = 0
+    no_neg_examples = False
+    for all_src, all_tgt in iterable:
```

Comment:
In general I feel like we're starting to have a lot of ambiguous types and wondering if we shouldn't subclass for your case, rather than trying to bring more logic into SLM?

Reply:
Thanks for the suggestions. I think adding negative examples would be in general useful (if not now, later) incase if we implement different losses that consider negative examples as well. So, I feel like SFLM shud have this capability of processing negatives. Probably, the current implementation is not the best to add this feature, but we will improve it later before merging into main branch. Currently, moving on to unblock some experiments.